spec/features/admin/firm_approval_spec.rb

```diff
@@ -1,18 +1,22 @@
 RSpec.feature 'Approving firms on the admin interface', :inline_job_queue do
```

Comment:
This spec is failing when I run this branch locally, and I think it's a pretty good illustration of some of the issues we're introducing with these changes.

Primarily, this spec doesn't really accurately reflect the behaviour of the system anymore. If we have "fully_registered_principal_users", the way the code's written means that we expect an API-based verification attempt to be triggered in a callback at the point that a principal is created. In switching off the TestAfterCommit config (elsewhere in this PR), two things have happened:

- The code under test doesn't behave like it does in non-test environments unless we explicitly execute the after_commit callbacks.
- We're relying on the fact that those callbacks aren't run to make sure that the database is in the right state for us to be able to click a verify button manually later on in the scenario. This is very hidden and makes following the test harder.

These issues are compounded by the fact that the `verify_fca_number` callback added to `Principal` creates an infinite loop in its current form, and needs to changed to update the column directly (or some other solution).

I'm going to have a crack at reworking some of this to address the issues above, will commit directly to this branch.

Reply:
@malcolmbaig good shout